packages/authorization-process/test/api/setAdminToClient.test.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { generateToken, getMockClient } from "pagopa-interop-commons-test";
+import { authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { generateId, TenantId, UserId } from "pagopa-interop-models";
+import { authorizationApi } from "pagopa-interop-api-clients";
+import { api, authorizationService } from "../vitest.api.setup.js";
+import { clientToApiClient } from "../../src/model/domain/apiConverter.js";
+import {
+  clientAdminAlreadyAssignedToUser,
+  clientKindNotAllowed,
+  clientNotFound,
+  userWithoutSecurityPrivileges,
+} from "../../src/model/domain/errors.js";
+
+describe("API POST /clients/{clientId}/admin test", () => {
+  const mockClient = getMockClient();
+  const body = { adminId: generateId<UserId>() };
+  const makeRequest = async (token: string, clientId: string = mockClient.id) =>
+    request(api)
+      .post(`/clients/${clientId}/admin`)
+      .set("Authorization", `Bearer ${token}`)
+      .set("X-Correlation-Id", generateId())
+      .send(body);
+
+  const apiResponse = authorizationApi.Client.parse(
+    clientToApiClient(mockClient, { showUsers: false })
+  );
+
+  beforeEach(() => {
+    authorizationService.setAdminToClient = vi
+      .fn()
+      .mockResolvedValue({ client: mockClient, showUsers: false });
+  });
+
+  it("Should return 200 for user with role Admin", async () => {
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(200);
+    expect(res.body).toEqual(apiResponse);
+  });
+
+  it.each([
+    {
+      description: "invalid clientId",
+      clientId: "invalid-client-id",
+      body,
+    },
+    { description: "invalid body: {}", clientId: mockClient.id, body: {} },
+    {
+      description: "invalid body: { adminId: 123 }",
+      clientId: mockClient.id,
+      body: { adminId: 123 },
+    },
+    {
+      description: "invalid body: { adminId: '' }",
+      clientId: mockClient.id,
+      body: { adminId: "" },
+    },
+    {
+      description: "invalid body: { invalidField: 'foo' }",
+      clientId: mockClient.id,
+      body: { invalidField: "foo" },
+    },
+  ])(
+    "Should return 400 if passed an $description",
+    async ({ clientId, body }) => {
+      const token = generateToken(authRole.ADMIN_ROLE);
+      const res = await request(api)
+        .post(`/clients/${clientId}/admin`)
+        .set("Authorization", `Bearer ${token}`)
+        .set("X-Correlation-Id", generateId())
+        .send(body);
```

Comment:
```suggestion
    "Should return 400 if passed invalid params: %s",
    async ({ clientId, adminId }) => {
      const token = generateToken(authRole.ADMIN_ROLE);
      const res = await makeRequest(token, clientId as ClientId, adminId);
```
I would align this with the other tests by not using `description` inside the `it.each`; moreover you can reuse `makeRequest` by passing `adminId`

Reply:
I don't think I can use `makeRequest`, as suggested, which takes `adminId` as the third parameter instead of the _body_, because I wouldn't be able to test the case where the body is invalid anymore.
**What do you think about?**

The following is the refactor

```suggestion
    it.each([
    {
      clientId: "invalid-client-id",
      adminId: generateId<UserId>(),
    },
    { description: "invalid body: {}", clientId: mockClient.id, body: {} },
    {
      clientId: mockClient.id,
      adminId: 123 as unknown as UserId,
    },
    {
      clientId: mockClient.id,
      adminId: "" as UserId,
    },
    {
      clientId: mockClient.id,
      adminId: "foo" as UserId,
    },
  ])(
    "Should return 400 if passed invalid params: %s",
    async ({ clientId, adminId }) => {
      const token = generateToken(authRole.ADMIN_ROLE);
      const res = await makeRequest(token, clientId as ClientId, adminId);
      expect(res.status).toBe(400);
    }
  );
```

Reply:
You can put `as UserId` here --> makeRequest(token, clientId as ClientId, adminId `as UserId`); 
and remove it from other parts

Reply:
Much better